tests/pytests/test_aggregate.py

```diff
@@ -1032,3 +1032,50 @@ def testGroupAfterSort(env):
     #    one for (t == AAAA, n == 0), one for (t == BBBB, n == 0), and one for (t == ????, n == 1)
 
     env.assertEqual(res, expected)
+
+
+def testWithKNN(env):
```

Comment:
consider adding a test with no `SORTBY`, and a hybrid case. Also maybe a `RANGE` sanity?